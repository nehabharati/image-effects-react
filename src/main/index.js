import styled from 'styled-components';
import { filterList } from './css/filterList';

export const ImageContainer = styled.figure`
  width: 100%;
  height: 15rem;
  margin: 0;

  ${(props) => {
    let {
      saturation,
      brightness,
      contrast,
      sepia,
      grayscale,
      hueRotate,
      filter,
      backgroundBefore,
      backgroundAfter,
    } = props.options;

    return filterList[filter]({
      saturation,
      brightness,
      contrast,
      sepia,
      grayscale,
      hueRotate,
      backgroundBefore,
      backgroundAfter,
    });
  }}
`;

ImageContainer.defaultProps = {
  options: {
    saturation: 1.6,
    brightness: 1.1,
    contrast: undefined,
    sepia: 0.3,
    grayscale: undefined,
    hueRotate: '-10deg',
    filter: 'walden',
    backgroundBefore: undefined,
    backgroundAfter: undefined,
  },
};
