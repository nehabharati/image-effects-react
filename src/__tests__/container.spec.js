import React from 'react';
import { ImageContainer } from '../main/index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('No options', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ImageContainer>
          <img src="" />
        </ImageContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ImageContainer block', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ImageContainer options={{ filter: 'walden' }}>
          <img src="" />
        </ImageContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('add property', () => {
  it('renders with right property', () => {
    const tree = renderer
      .create(
        <ImageContainer
          options={{
            filter: 'walden',
            hueRotate: '-20deg',
            contrast: '2',
            sepia: '0.5',
            saturate: '1',
          }}
        >
          <img src="" />
        </ImageContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
