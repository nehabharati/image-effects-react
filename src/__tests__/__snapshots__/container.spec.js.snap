// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer block renders correctly 1`] = `
.c1 {
  width: 100%;
  height: 15rem;
  margin: 0;
  position: relative;
}

.c1 img {
  width: 100%;
  height: 15rem;
  object-fit: cover;
}

-webkit-filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6)  filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6) .c0:before {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 2;
}

.c1:after {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 3;
}

.c1::after {
  background: #0044cc;
  mix-blend-mode: screen;
  opacity: .3;
}

<figure
  className="c0 c1"
>
  <img
    src=""
  />
</figure>
`;

exports[`No options renders correctly 1`] = `
.c1 {
  width: 100%;
  height: 15rem;
  margin: 0;
  position: relative;
}

.c1 img {
  width: 100%;
  height: 15rem;
  object-fit: cover;
}

-webkit-filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6)  filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6) .c0:before {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 2;
}

.c1:after {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 3;
}

.c1::after {
  background: #0044cc;
  mix-blend-mode: screen;
  opacity: .3;
}

<figure
  className="c0 c1"
>
  <img
    src=""
  />
</figure>
`;

exports[`No optionss renders correctly 1`] = `
.c1 {
  width: 100%;
  height: 15rem;
  margin: 0;
  position: relative;
}

.c1 img {
  width: 100%;
  height: 15rem;
  object-fit: cover;
}

-webkit-filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6)  filter:  hue-rotate(-10deg)  brightness(1.1)   sepia(0.3) saturate(1.6) .c0:before {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 2;
}

.c1:after {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 3;
}

.c1::after {
  background: #0044cc;
  mix-blend-mode: screen;
  opacity: .3;
}

<figure
  className="c0 c1"
>
  <img
    src=""
  />
</figure>
`;

exports[`add property renders with right property 1`] = `
.c1 {
  width: 100%;
  height: 15rem;
  margin: 0;
  position: relative;
}

.c1 img {
  width: 100%;
  height: 15rem;
  object-fit: cover;
}

-webkit-filter:  hue-rotate(-20deg)  brightness(1.1)   sepia(0.5) saturate(1.6)  filter:  hue-rotate(-20deg)  brightness(1.1)   sepia(0.5) saturate(1.6) .c0:before {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 2;
}

.c1:after {
  content: '';
  display: block;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
  pointer-events: none;
  z-index: 3;
}

.c1::after {
  background: #0044cc;
  mix-blend-mode: screen;
  opacity: .3;
}

<figure
  className="c0 c1"
>
  <img
    src=""
  />
</figure>
`;
